redhat/build.pkr.hcl

```diff
@@ -14,21 +15,59 @@ build {
     ]
 
     execute_command   = "echo '${local.username}' | {{ .Vars }} sudo -S -E sh -eux '{{ .Path }}'"
+    except            = var.redhat_username == "" || var.redhat_password == "" ? ["parallels-iso.image"] : []
+  }
+
+  ## Miscellaneous setup items
+  provisioner "shell" {
+    scripts = [
+      "${path.root}/../scripts/rhel/base/tmpdir.sh",
+      "${path.root}/../scripts/rhel/base/sshd.sh",
+    ]
+
+    execute_command   = "echo '${local.username}' | {{ .Vars }} sudo -S -E sh -eux '{{ .Path }}'"
+    expect_disconnect = false
+  }
+
+  ## Update all OS Packages
+  ## Requires Red Hat credentials
+  provisioner "shell" {
+    scripts = [
+      "${path.root}/../scripts/rhel/base/update.sh",
+    ]
+
+    execute_command   = "echo '${local.username}' | {{ .Vars }} sudo -S -E sh -eux '{{ .Path }}'"
+    except            = var.redhat_username == "" || var.redhat_password == "" ? ["parallels-iso.image"] : []
     expect_disconnect = true
+  }
+
+  ## Install Parallels Tools
```

Comment:
Installing the Parallels tools before the UI will have some downside effects. One is the lack of copy/paste between the machines. This behaviour is at least reproducible in the Ubuntu distributions, and RedHat should not be any different.

What we can do is move the PD Tools after the desktop is enabled and a reboot cycle has been done.

Can you confirm with this setup you can still copy/paste into the machine?

Reply:
I will do whatever you want here, but you bring up an interesting point anyway: when we select to install the desktop we do it twice! First we add it to the kickstart file with `provisioner.pkr.hcl` (look at `http_content`), and then we install the exact same thing with `desktop.sh`.

The Tools installation was always before the `desktop.sh` script, but the desktop was already installed in the %packages section of the kickstart file. I think that's why that ever worked at all...

So we have three places we *could* install the Desktop: kickstart, `desktop.sh`, or we could add it as part of the add-on processing (as long as it doesn't need a reboot).

So the quickest thing is to just remove the `desktop.sh` provisioner, because right now it's redundant.

Adding packages to the kickstart %packages section is actually pretty nifty, because everything we add there that is on the installation media (which is actually that done in addons except docker, flutter, and vscode) is installed from the RHEL media, and doesn't require Red Hat credentials unless we want to update.

Reply:
@stephenulmer you bring a good point, lets do that then. Removing the dekstop.sh